tests/e2e/nvidia-ctk_containerd_test.go

```diff
@@ -0,0 +1,433 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// nvidiaCtkConfigureContainerdCmd is the command template to configure
+// containerd using nvidia-ctk.
+// Placeholders: %s for additional flags (e.g., --nvidia-set-as-default)
+var nvidiaCtkConfigureContainerdCmd = `nvidia-ctk runtime configure --runtime=containerd --config=/etc/containerd/config.toml --drop-in-config=/etc/containerd/config.d/99-nvidia.toml %s`
+
+// Integration tests for containerd drop-in config functionality
+var _ = Describe("containerd", Ordered, ContinueOnFailure, func() {
+	var (
+		nestedContainerRunner Runner
+		// docker.io/kindest/base:v20250521-31a79fd4 comes with
+		// containerd v2.1.1
+		outerContainerImage     = "docker.io/kindest/base:v20250521-31a79fd4"
+		containerName           = "nvctk-e2e-nvidia-ctk-containerd-tests"
+		nestedContainerCacheDir string
+	)
+
+	BeforeAll(func(ctx context.Context) {
+		var err error
+
+		// Create a tempdir on the remote runner (not nested container)
+		var tmpdir string
+		tmpdir, _, err = runner.Run("mktemp -d --tmpdir=/tmp nvctk-e2e-test-cacheXXX")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(strings.TrimSpace(tmpdir)).ToNot(BeEmpty())
+		nestedContainerCacheDir = strings.TrimSpace(tmpdir)
+
+		// Prepare cache on the remote runner (where Docker is)
+		installer, err := NewToolkitInstaller(
+			WithToolkitImage(nvidiaContainerToolkitImage),
+			WithCacheDir(nestedContainerCacheDir),
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, _, err = installer.PrepareCache(runner)
+		Expect(err).ToNot(HaveOccurred(), "Failed to prepare cache for containerd")
+
+		// Now create the nested container with the cache mounted
+		nestedContainerRunner, err = NewNestedContainerRunner(runner, outerContainerImage, installCTK, containerName, nestedContainerCacheDir)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Backup original containerd configuration
+		_, _, err = nestedContainerRunner.Run(`
+			# Backup the original config.toml
+			if [ -f /etc/containerd/config.toml ]; then
+				cp /etc/containerd/config.toml /tmp/containerd-config.toml.backup
+			fi
+			
```

Comment:
Just as a thought. Does using `-f -r` as arguments for `cp` for all of the things we want to backup make this (and the subsequent cleanup) simpler?